frame/evm/src/runner/stack.rs

```diff
@@ -1170,6 +1183,13 @@ where
 			code.len(),
 			address
 		);
+
+		// The EVM returns empty code in case of an EIP-7702 delegation to the zero address.
+		// In response we must remove such delegation from the authorizing account.
+		if code.is_empty() {
+			Pallet::<T>::remove_account_code(&address);
+		}
+
 		Pallet::<T>::create_account(address, code, caller)
```

Comment:
I think this should be:
```suggestion
		if code.is_empty() {
			Pallet::<T>::remove_account_code(&address);
		} else {
		    Pallet::<T>::create_account(address, code, caller)
		}
```
If we look at the `create_account` function, it already returns early if `code.is_empty() == true`

Reply:
This will be addressed after the changes to `evm` are merged: https://github.com/rust-ethereum/evm/pull/367